packages/@uppy/companion/src/server/provider/zoom/index.js

```diff
@@ -210,30 +212,43 @@ class Zoom extends Provider {
         modifiedDate: adapter.getDateFolderModified(end),
         size: null
       })
-      end = start.clone().subtract(1, 'days')
-      // if the limit is mid-month, keep that exact date
-      start = (end.isSame(limit, 'month') && limit.date() !== 1) ? limit.clone() : end.clone().date(1)
+      end = start.clone().subtract(1, 'days').endOf('day')
+      start = (end.isSame(limit, 'month') && limit.date() !== 1) ? limit.clone() : end.clone().date(1).startOf('day')
     }
-    data.nextPagePath = allResultsShown ? null : adapter.getDateNextPagePath(end.clone())
+    data.nextPagePath = allResultsShown ? null : adapter.getDateNextPagePath(end)
     return data
   }
 
-  _adaptData (userResponse, results) {
+  _adaptData (userResponse, results, query) {
     if (!results) {
       return { items: [] }
     }
 
+    // convert start limit from local time to UTC
+    const utcFrom = moment.tz(query.from, userResponse.timezone || 'UTC').startOf('day').tz('UTC')
+    const utcTo = moment.tz(query.to, userResponse.timezone || 'UTC').endOf('day').tz('UTC')
+
     const data = {
       nextPagePath: adapter.getNextPagePath(results),
       items: [],
       username: adapter.getUserEmail(userResponse)
     }
-    const items = results.meetings || results.recording_files.filter(file => file.file_type !== 'TIMELINE')
+
+    let items = []
+    if (results.meetings) {
+      items = results.meetings
+        .map(item => { return { ...item, utcStart: moment.utc(item.start_time) } })
+        .filter(item => moment.utc(item.start_time).isAfter(utcFrom) && moment.utc(item.start_time).isBefore(utcTo))
```

Comment:
This filters out extra results, should account for cases below:

case 1: user in PST (UTC-07:00) --> wants 08-01 to 08-31 (PST), API call for 08-01 to 09-01 (UST), filter out items starting after 09-01 06:59 UST (08-31 23:59 PST)
case 2: user in JST (UTC+09:00) --> wants 08-01 to 08-31 (KST), API call for 07-31 to 08-31 (UST), filter out items starting before 07-31 15:00 UST (08-01 00:00 JST)